core/src/main/java/brooklyn/catalog/internal/CatalogInitialization.java

```
@@ -0,0 +1,380 @@
+        synchronized (populatingCatalogMutex) {
+            try {
+                isPopulating = true;
+                BasicBrooklynCatalog catalog = (BasicBrooklynCatalog) managementContext.getCatalog();
+                if (!catalog.getCatalog().isLoaded()) {
+                    catalog.load();
+                } else {
+                    if (needsInitial && (hasRunOfficial || hasRunBestEffort)) {
+                        // an indication that something caused it to load early; not severe, but unusual
+                        log.warn("Catalog initialization has not properly run but management context has a catalog; re-populating, possibly overwriting items installed during earlier access (it may have been an early web request)");
+                        catalog.reset(ImmutableList.<CatalogItem<?,?>>of());
+                    }
+                }
+                hasRunOfficial = true;
+
+                populateCatalogImpl(catalog, needsInitial, optionalItemsForResettingCatalog);
+            } finally {
+                hasRunOfficial = true;
+                isPopulating = false;
+            }
+        }
+    }
+
+    private void populateCatalogImpl(BasicBrooklynCatalog catalog, boolean needsInitial, Collection<CatalogItem<?, ?>> optionalItemsForResettingCatalog) {
+        applyCatalogLoadMode();
+        
+        if (optionalItemsForResettingCatalog!=null) {
+            catalog.reset(optionalItemsForResettingCatalog);
+        }
+        
+        if (needsInitial) {
+            populateInitial(catalog);
+        }
+        
+        populateAdditions(catalog);
+
+        populateViaCallbacks(catalog);
+    }
+
+    private enum PopulateMode { YAML, XML, AUTODETECT }
+    
+    protected void populateInitial(BasicBrooklynCatalog catalog) {
+        if (disallowLocal) {
+            if (!hasRunOfficial()) {
+                log.debug("CLI initial catalog not being read with disallow-local mode set.");
+            }
+            return;
```

If the user has set `initialUri`, but at the same time has `CATALOG_LOAD_MODE=LOAD_PERSISTED_STATE` configured, I'd expect the new style to take precedence. Or even fail early and refuse to start with a descriptive message.
Letting the old style win with only a debug message is confusing.